src/Symfony/Component/Debug/Tests/ExceptionHandlerTest.php

```diff
@@ -89,6 +90,43 @@ public function testHandle()
         $handler->handle($exception);
     }
 
+    public function testPsr7()
+    {
+        $exception = FlattenException::create(new \Exception('foo'), 123, array('Header' => 'Value'));
+
+        $body = $this->getMock('Psr\Http\Message\StreamInterface');
+        $body
+            ->expects($this->exactly(1))
+            ->method('write')
+            ->with($this->stringContains('Whoops, looks like something went wrong.'));
+
+        $response = $this->getMock('Psr\Http\Message\ResponseInterface');
+        $response
+            ->expects($this->exactly(1))
+            ->method('withStatus')
+            ->with(123)
+            ->will($this->returnValue($response));
```

Comment:
returning the same mock for the response does not match the PSR-7 behavior, and it means that the test will still pass if you remove one of the assignments of the modified response (or even all of them)